### PR TITLE
[MAINT] Apply precommit hooks to doc

### PR DESCRIPTION
**Pull request by Remi-Gau**
Relates to #2528 

Changes proposed in this pull request:

- apply pre-commit hooks to doc
- manual fixes of PEP8 errors
- adapt flake8 config to silence certain errors specific to the doc
- omits doc files involved in other PRs (see listing here)
- minor manual refactoring in conf.py

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
 
-files: "(nilearn/de.*)|(.*binder/)|(.*circleci/)|(.*github/)|(maint_tools/.*)|(setup.*)|(.*coverage)|(.*toml$)|(.*in$)|(.*yml)|(.*yaml)|(CONTRIBUTING.rst)|(AUTHORS.rst)|(LICENSE)|(Makefile)|(.*gitignore)"
+files: "(nilearn/de.*)|(.*binder/)|(.*circleci/)|(.*github/)|(maint_tools/.*)|(setup.*)|(.*coverage)|(.*toml$)|(.*in$)|(.*yml)|(.*yaml)|(CONTRIBUTING.rst)|(AUTHORS.rst)|(LICENSE)|(Makefile)|(.*gitignore)|(doc/.*)"
+exclude: "^(doc/whats_new.rst)|(doc/changes/(names|latest|0\\.9\\.[0-2]|0\\.10\\.0).rst)|(doc/(glm/first_level_model\\.rst|introduction\\.rst|maintenance\\.rst|manipulating_images/input_output\\.rst|modules/datasets\\.rst|modules/reference\\.rst|references\\.bib|themes/nilearn/static/nature\\.css_t))$"
```
Comment (Remi-Gau) [inline .pre-commit-config.yaml:5]: Will probably lead to some merge conflicts with the PR for reformatting of the example, will handle those in due time.

**Comment by Remi-Gau** [comment]
Even though the changes are not big, it is probably worth an entry in the change log, no?

**Comment by Remi-Gau** [comment]
Actually given this one changes mostly trailing white spaces and blank lines at the end of files, I suspect it can be merged before #3556

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks !

**Review by ymzayek — APPROVED** [review]
LGTM thx!

**Comment by Remi-Gau** [comment]
merging

**Pull request merged**